src/borg/platform/linux.pyx

```diff
@@ -8,11 +8,25 @@ from ..helpers import user2uid, group2gid
 from ..helpers import safe_decode, safe_encode
 from .base import SyncFile as BaseSyncFile
 from .base import safe_fadvise
+from .xattr import _listxattr_inner, _getxattr_inner, _setxattr_inner, split_string0
 
 from libc cimport errno
 from libc.stdint cimport int64_t
 
-API_VERSION = '1.1_03'
+API_VERSION = '1.2_01'
+
+cdef extern from "attr/xattr.h":
+    ssize_t c_listxattr "listxattr" (const char *path, char *list, size_t size)
+    ssize_t c_llistxattr "llistxattr" (const char *path, char *list, size_t size)
+    ssize_t c_flistxattr "flistxattr" (int filedes, char *list, size_t size)
+
+    ssize_t c_getxattr "getxattr" (const char *path, const char *name, void *value, size_t size)
+    ssize_t c_lgetxattr "lgetxattr" (const char *path, const char *name, void *value, size_t size)
+    ssize_t c_fgetxattr "fgetxattr" (int filedes, const char *name, void *value, size_t size)
+
+    int c_setxattr "setxattr" (const char *path, const char *name, const void *value, size_t size, int flags)
```

Comment:
the declaration for listxattr, getxattr, setxattr is missing completely in the ctypes code:

https://github.com/borgbackup/borg/pull/3918/files#diff-87630d54442e9151262050bf3d4bad49L192